### PR TITLE
[WIP] [FIX] Regression introduced by #4936

### PR DESCRIPTION
**Pull request by massich**
assert_allclose(data1, data2)
see
https://github.com/mne-tools/mne-python/pull/5591#issuecomment-427772424

### CODE REVIEW

File: mne/io/eeglab/eeglab.py [hunk 1]
```diff
@@ -812,3 +812,12 @@ def read_annotations_eeglab(fname, uint16_codec=None):
...
     return int("".join([x for x in trigger if x.isdigit()]))
+
+
+def _strip_to_integer_new(trigger):
+    """Return only the integer part of a string."""
+    trigger = "".join([x for x in trigger if x.isdigit()])
+    if trigger.isdigit():
+        return int(trigger)
+    else:
+        return None
```
Comment (jona-sassenhagen) [inline mne/io/eeglab/eeglab.py]: This is done with a regex in the BV reader, isn't it?

**Comment by jona-sassenhagen** [comment]
looks good to me ...

**Comment by cbrnr** [comment]
When did the old `_strip_to_integer` function fail?

**Comment by massich** [comment]
@cbrnr it is not that the `_strip_to_integer` failed (perse) is that some of the functionality in `read_events_{eeglab|brainvsion|xxx}` drop events. This right now is taken care by `event_id_func` so `_strip_to_integer` alone cannot take care of it.

I know, the naming right now is far from ideal. But this PR is just a hotfix, needed in #5591. Once we don't use deprecation code, I intend to improve a bit whatever is left. And review all the namings (and deprecate the old ones) once the behavior of all readers is more or less unified. 

So, if you think that its good enough. Please merge.

**Pull request closed without merging**